core/builder/src/main/java/io/quarkus/builder/JsonTransform.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.builder;
+
+import java.util.function.Predicate;
+
+@FunctionalInterface
+public interface JsonTransform {
```

Comment:
I think `Transform` is the better name. That's the name of the function (as annotated by the `@FunctionalInterface`), it transforms. Btw, the transform part is heavily inspired by the upcoming [Class-File API](https://openjdk.org/jeps/457) where they use `Transform` instead of `Transformer`.